### PR TITLE
Add an Info.plist flag to enable MacOS platform views preview support

### PR DESCRIPTION
**Pull request by RichardJCai**
Enabling the flag `io.flutter_embedded_views_preview` enables platform view support for MacOS.

Additionally, when the flag is enabled, the render and main threads are merged, however this disables smooth resizing as smooth resizing requires synchronization and blocking between the main and render thread.

Tentatively using the `io.flutter_embedded_views_preview` flag as it was used in #6756. 

Please only review the last commit, this PR depends on https://github.com/flutter/engine/pull/22905 being merged.

### CODE REVIEW

File: shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm [hunk 1]
```diff
@@ -4,20 +4,29 @@
 
 #include "flutter/fml/logging.h"
 
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterConstants.h"
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController_Internal.h"
 
-@implementation FlutterPlatformViewController
+@implementation FlutterPlatformViewController {
+  bool _embeddedViewsEnabled;
```
Comment (cyanglaz) [inline shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm:11]: We don't need this instance method.

File: shell/platform/darwin/macos/framework/Source/FlutterConstants.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// The name of the Info.plist flag to enable the embedded MacOS views preview.
+const char* const kEmbeddedViewsPreview = "io.flutter_embedded_views_preview";
```
Comment (cyanglaz) [inline shell/platform/darwin/macos/framework/Source/FlutterConstants.h:6]: Don't need the constant in a separate file, can move it to `FlutterPlatformViewController.mm`

File: shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm [hunk 3]
```diff
@@ -56,6 +71,11 @@ - (void)onDisposeWithViewId:(int64_t)viewId result:(nonnull FlutterResult)result
 
 - (void)registerViewFactory:(nonnull NSObject<FlutterPlatformViewFactory>*)factory
                      withId:(nonnull NSString*)factoryId {
+  if (!_embeddedViewsEnabled) {
```
Comment (cyanglaz) [inline shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm:74]: Can just be if (![self embeddedViewsEnabled])

File: shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm [hunk 4]
```diff
@@ -4,20 +4,29 @@
...
-  self->_platformViewFactories = [[NSMutableDictionary alloc] init];
+  _platformViewFactories = [[NSMutableDictionary alloc] init];
+  [self setEmbeddedViewsEnabled:[FlutterPlatformViewController embeddedViewsEnabled]];
   return self;
 }
 
 - (void)onCreateWithViewId:(int64_t)viewId
                   viewType:(nonnull NSString*)viewType
                     result:(nonnull FlutterResult)result {
+  if (!_embeddedViewsEnabled) {
```
Comment (cyanglaz) [inline shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm:25]: I don't think we need this, the `if (factory == nil)` already handles this. 
Same in `onDispose`

File: shell/platform/darwin/macos/framework/Source/FlutterView.mm [hunk 5]
```diff
@@ -15,6 +16,7 @@ @interface FlutterView () <FlutterResizeSynchronizerDelegate> {
   __weak id<FlutterViewReshapeListener> _reshapeListener;
   FlutterResizeSynchronizer* _resizeSynchronizer;
   FlutterSurfaceManager* _surfaceManager;
+  bool _embedded_views_preview_enabled;
```
Comment (cyanglaz) [inline shell/platform/darwin/macos/framework/Source/FlutterView.mm:19]: Don't need this ivar, can just use [FlutterPlatformViewController embeddedViewsEnabled] when necessary.

**Review by cyanglaz — COMMENTED** [review]
I'm not sure about `[FlutterView present]` and `[FlutterView reshaped]`
Maybe @iskakaushik has more context as he reviewed https://github.com/flutter/engine/pull/21525?

cc @knopp

**Comment by knopp** [comment]
First iteration of smooth resizing worked with merged platform / raster threads so it should be possible to do it here as well. I'll look into it when this lands. Might be bit trickier for dynamic thread merging.

File: shell/platform/darwin/macos/framework/Source/FlutterView.mm (hunk 5 above)
Comment (knopp) [reply shell/platform/darwin/macos/framework/Source/FlutterView.mm:19]: ivars should be camelCase.

**Comment by chinmaygarde** [comment]
@iskakaushik Do you know who the owner for this is?

**Comment by RichardJCai** [comment]
> @iskakaushik Do you know who the owner for this is?

I'd be happy to continue working on this one, but waiting on this https://github.com/flutter/engine/pull/22905 to get reviewed.

**Pull request closed without merging**